File: c/PE029.c
#include <stdio.h>
#include <math.h>

int main() {
    double answer[10000];
    int i = 0;
    for (double a = 2; a<=100;a++)
        for (double b = 2;b<=100;b++) {
            double temp = pow(a,b);
            int j;
            for(j = 0; j < i; j++) {
                if (answer[j] == temp)
                    break;
            }
            if (j == i)
                answer[i++] = temp;
        }
    printf("%d\n", i);
    return 0;
}

File: java/PE029.java
import java.util.HashSet;
import java.math.BigInteger;

public class PE029 {
    public static void main(String[] args) {
        HashSet<BigInteger> set = new HashSet<BigInteger>();
        for(int a = 2; a <= 100; a++) {
            for(int b = 2; b <= 100; b++) {
                BigInteger big = new BigInteger("" + a + "");
                big = big.pow(b);
                set.add(big);
            }
        }
        System.out.println(set.size());
    }
}
